src/main/scala/org/clulab/wm/eidos/context/TimeNormFinder.scala

```diff
@@ -30,149 +43,187 @@ object TimeNormFinder {
-        val sentenceStart = sentence.startOffsets.head
-        val sentenceText = text.substring(sentenceStart, sentence.endOffsets.last)
-
-        // find all the places where a regular expression matches
-        val matches = timeRegexes.flatMap(_.findAllMatchIn(sentenceText))
-
-        // add some context around each match
-        val singleIntervals = for (m <- matches) yield {
-
-          // expand to the specified context window size
-          val start1 = math.max(0, m.start - CONTEXT_WINDOW_SIZE)
-          val end1 = math.min(sentenceText.length, m.end + CONTEXT_WINDOW_SIZE)
-          //assert(0 <= start1 && start1 <= m.start)
-          //assert(m.end <= end1 && end1 <= sentenceText.length)
-
-          // do not include context beyond 2 consecutive newline characters
-          val separator = "\n\n"
-          val start2 = sentenceText.slice(start1, m.start).lastIndexOf(separator) match {
-            case -1 => start1
-            case i => start1 + i + separator.length
-          }
-          val end2 = sentenceText.slice(m.end, end1).indexOf(separator) match {
-            case -1 => end1
-            case i => m.end + i
-          }
-          //assert(start1 <= start2 && start2 <= m.start)
-          //assert(m.end <= end2 && end2 <= end1)
-
-          // expand to word boundaries
-          def nextOrElse(iterator: Iterator[Int], x: => Int): Int = if (iterator.hasNext) iterator.next else x
-          val start3 = nextOrElse(sentence.startOffsets.reverseIterator.map(_ - sentenceStart).dropWhile(_ > start2), 0)
-          val end3 = nextOrElse(sentence.endOffsets.iterator.map(_ - sentenceStart).dropWhile(_ < end2), sentenceText.length)
-          //assert(0 <= start3 && start3 <= start2)
-          //assert(end2 <= end3 && end3 <= sentenceText.length)
-
-          //assert(start3 <= end3)
-          // yield the context interval adjusting the span to full-text character offsets
-          Interval(sentenceStart + start3, sentenceStart + end3)
-        }
+  def parseDctString(dctString: String): Option[DCT] = {
+    val timeExpressions: Array[TimeExpression] = parser.parse(dctString)
+    val dctOpt = timeExpressions.headOption.map { timeExpression =>
+      // use a SimpleInterval (character span = None) so it doesn't interfere with character span of TimeExpressions
+      val timeExInterval = timeExpression.asInstanceOf[TimExInterval] // Is this a leap of faith?
+      val simpleInterval = SimpleInterval(timeExInterval.start, timeExInterval.end)
 
-        // merge overlapping contexts
-        val mergedIntervals = singleIntervals.sorted.foldLeft(Seq.empty[Interval]) {
-          case (Seq(), context) => Seq(context)
-          case (init :+ last, context) if last.end > context.start => init :+ Interval(last.start, context.end)
-          case (contexts, context) => contexts :+ context
-        }
+      DCT(simpleInterval, dctString)
+    }
```

Comment:
Yes, that's a leap of faith. There's no guarantee it returns an `Interval`. For example, if the DCT string is just "weeks", it will return a `RepeatingInterval`. I would write this instead as something like:
```
parser.parse(dctString).collectFirst { case i: TimExInterval => DCT(SimpleInterval(i.start, i.end), dctString) }
```
Depending on how careful you want to be, you may also want to consider a `Try` around that `.parse`, since in some weird situations the machine learning model produces an invalid composition of time operators. I think @EgoLaparra has an example of where this happens, since he had to add a `Try` in `parseBatch`.

Reply:
In my case, I found that`Season-Of-Year` entities like "rainfall" produce compositions that are not necessarily invalid but cannot be interpreted. However, what @bethard says can happen if the parser misses the correct values of the components of the time expression.

Reply:
Great catch!  The headOption took care of one exception, but if more are possible, I should deal with them here.

Reply:
An example that could occur as a DCT: `18 Maya 2018`. The parser identifies `Maya` as a `Month-Of-Year` but is obviously a wrong value.

Reply:
I added a Try and pushed the changes.  There will be at least one more push which updates the processors version.  You comments have been helpful, so please continue.